Tests/MapboxCoreNavigationTests/NativeHandlersFactoryTests.swift

```diff
@@ -0,0 +1,82 @@
+import XCTest
+import MapboxNavigationNative
+import MapboxDirections
+@testable import MapboxCoreNavigation
+
+private let customConfigKey = "com.mapbox.navigation.custom-config"
+
+class ConfigFactorySpy: ConfigFactory {
+    
+    static var passedCustomConfig: String?
+    
+    override class func build(for profile: SettingsProfile,
+                              config: NavigatorConfig,
+                              customConfig: String) -> ConfigHandle {
+        passedCustomConfig = customConfig
+        return super.build(for: profile, config: config, customConfig: customConfig)
+    }
+}
+
+class NativeHandlersFactoryTests: XCTestCase {
+    
+    var handlersFactory: NativeHandlersFactory!
+    
+    override func setUp() {
+        UserDefaults.standard.set(nil, forKey: customConfigKey)
+        handlersFactory = NativeHandlersFactory(tileStorePath: "tile store path",
+                                                credentials: DirectionsCredentials(),
```

Comment:
You can use top level: `directions` of `DirectionsSpy` type.

Reply:
Thanks, I ended up using `Fixture.credentials`